numpy/core/memmap.py

```diff
@@ -268,14 +266,13 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
             self.offset = offset
             self.mode = mode
 
-            if isinstance(filename, basestring):
-                self.filename = os.path.abspath(filename)
-            elif is_pathlib_path(filename):
+            if is_pathlib_path(filename):
+                # special case - if we were constructed with a pathlib.path,
+                # then filename is a path object, not a string
                 self.filename = filename.resolve()
```

Comment:
This is the only remaining use of `is_pathlib_path`, which is needed for a weird special case.

Reply:
Since we need `is_pathlib_path` anyway, PR #11348 more straight-forward and simpler.

Reply:
This is orthogonal to PR #11348 - this makes numpy work with backports of pathlib too / third-party objects implementing `__fspath__` too.